lib/format_parser.rb

```diff
@@ -30,24 +39,30 @@ def self.parse_http(url)
     parse(cached_io)
   end
 
-  def self.parse(io)
+  def self.parse(io, **opts, &proc)
```

Comment:
To be honest I'm not convinced of the block+opts approach. All of our code is Ruby2+ so we can use keyword arguments at will. Keyword arguments also give us a sane interface to specify default values. For example, we can use magic Symbols to signify a default with which you want all the natures:

```
def self.parse(io, accept_natures: :all, ...)
```

Should we want to use something as an argument for externally specified configuration everywhere, we can use any object that responds to `to_hash` and pass it in as the replacement for keyword arguments:

```
FormatParser.parse(io, **FORMAT_PARSER_CONFIG)
```

I.e. I do get the value of block configuration, but I don't see it in this use pattern. Moreover, we should be distrustful of the _file contents_, but not of the way how someone is using the library. We trust the user to make the right choices and if there is an incorrect file nature passed we can capture it at runtime - this will be a nature no parser supports.

A very valid example of block configuration is when it is divorced from the call site. For example:

```
SomeLibrary.configure |c|
  c.use_evil_corporate_http_proxy_from_microsoft = true
end

#... in another module that does something useful
SomeLibrary.perform_dispatch(name, title) # will indeed use the draconian proxy
```

but we are not doing it here and I don't think we might want to. @bermannoah how is the ATC usage pattern looking so far with regards to this?

Reply:
Agree on swapping back to keyword args. Since ATC uses FormatParser once per file parse job, it would end up using `accept_natures: :all` anyway.

Reply:
Hmmm, it's true that the block code looks superfluous right now. I think it would become handier when the number of options would wild up. I'll remove and kill dry-db dependency alongside.